### PR TITLE
[modbus] Add hints to potential problems on Modbus readout

### PR DESCRIPTION
**Pull request by mstormi**
Signed-Off-By: Markus Storm <markus.storm@gmx.net>

### CODE REVIEW

File: bundles/org.openhab.binding.modbus/README.md [hunk 1]
```diff
@@ -985,6 +988,18 @@ end
...
+## Troubleshooting
+
+Modbus, while simple at its heart, potentially is a complicated standard to use because there's a lot of freedom (and bugs) when it comes to implementations.
+There are many device or vendor specific quirks and wrinkles you might stumble across. Here's some:
+
+* a device may have more than one interface with a port open (502 or other) that it responds on, but may have no connection to the real bus hardware.
+Check ALL interfaces including (and preferring) any serial connection. Usually either that or the Ethernet IP will do.
+
+* some devices do not allow to query a range of registers that is too large or spans reserved registers.
+They may respond with an error or no error but invalid register data.
```
Comment (ssalonen) [inline bundles/org.openhab.binding.modbus/README.md:1000]: Can we make the troubleshooting more actionable?

For example, trying first to read large spans (more efficient especially with tens of registers with short polling), containing even undefined/undocumented data regions.

And if that does not work, trying to read the data in pieces that contains only documented/valid data regions.


We have seen all sorts of behavior 
- ok to query undefined registers, in the end you would only use the documented registers. 
- slave not responding at all if the data region contains undefined data. Will produce timeout error in binding 
- slave responding with explicit error, telling that addresses are invalid. Thing will go offline and description will tell the reason

The weirdest is where you need to query 4 registers, even if you are only needing the first two. This is an issue with SMA devices https://github.com/openhab/openhab1-addons/issues/4931#issue-197499556. Comes up every now and then as people stumble on it.

Comment (ssalonen) [reply bundles/org.openhab.binding.modbus/README.md:1000]: Perhaps the advice would be to review if any of the device specific modbus bindings are applicable for your device, e.g. Sunspec modbus binding. Many quirks are taken cared for and it's much easier for the user than the generic modbus binding.

I think the list is in the beginning of the document.

What do you think?

Comment (mstormi) [reply bundles/org.openhab.binding.modbus/README.md]: I've put up a couple of changes and hope they match your comments.

Regarding the last one, this doesn't belong into troubleshooting IMHO so I didn't do anything about it.
It should be more prominently emphasized than now at the beginning of the document though.

**Comment by Hilbrand** [comment]
Please be carefull when editing here on GitHub. Because you are a contributor you branch is created on the main openHAB-addons repository and not on your own fork.

**Comment by mstormi** [comment]
Ouch, I wasn't aware, sorry. Thanks for pointing out. Ok will delete after merge.
I was irritated already in the beginning when I used the Edit link on the page bottom and thought that goes into the docs repo
(as it usually does). @Confectrician is that intentional ?

**Comment by Hilbrand** [comment]
The addon readme's are mainainted with the addons. So it's correct that it opens this page. However, there is some unexpected behaviour of GitHub. When you edit a page on GitHub but are not a member of the repository it will automatically create the pr to your own fork. However, if you are a member of the repository it will by default create a branch on this repository.

File: bundles/org.openhab.binding.modbus/README.md [hunk 2]
```diff
@@ -174,11 +174,14 @@ You must give each of your bridge Things a reference (thing ID) that is unique f
...
 | `cacheMillis` | integer |          | `50`               | Duration for data cache to be valid, in milliseconds. This cache is used only to serve `REFRESH`  commands. Use zero to disable the caching.                                                   |
 
-Note: Polling can be manually triggered by sending `REFRESH` command to item bound to channel of `data` thing.
+Polling can be manually triggered by sending `REFRESH` command to item bound to channel of `data` thing.
 When manually triggering polling, a new poll is executed as soon as possible, and sibling `data` things (i.e. things that share the same `poller` bridge) are updated.
 In case the `poller` had just received a data response or an error occurred, a cached response is used instead.
 See [Refresh command](#refresh-command) section for more details.
 
+Some devices do not allow to query a range of registers that is too large or spans reserved registers.
+Eventually split your poller into multiple smaller ones to work around this problem.
```
[the remark this answers is not in the document]
Comment (Hilbrand) [reply bundles/org.openhab.binding.modbus/README.md:183]: Any changes here?

@ssalonen you can use the icon image to suggest an alternative text.

[the remark this answers is not in the document]
Comment (mstormi) [reply bundles/org.openhab.binding.modbus/README.md]: I reworded that

**Pull request merged**